src/features/WriteStoriesModal/WriteStoriesModal.tsx

```diff
@@ -2,27 +2,272 @@ import React from "react";
+import storyNameSidebarImg from "./assets/02-story-name-sidebar.png";
+import argsImg from "./assets/03-args.png";
+
+import dataJavascript from "./code/javascript";
+import dataTypescript from "./code/typescript";
+import dataTypescriptNextjs from "./code/nextjs-typescript";
+import { useGetProject } from "./hooks/useGetFrameworkName";
+import { API, AddonStore } from "@storybook/manager-api";
+
+export function WriteStoriesModal({
+  onFinish,
+  api,
+  addonsStore,
+}: {
+  onFinish: () => void;
+  api: API;
+  addonsStore: AddonStore;
+}) {
+  const [step, setStep] = React.useState<
+    "imports" | "meta" | "story" | "args" | "customStory"
+  >("imports");
+
+  const [isWarningStoryCopied, setWarningStoryCopied] = React.useState(false);
+
+  const buttonPath = useGetButtonPath();
+  const doesWarningButtonExist = useGetWarningButtonStatus(
+    step === "customStory",
+    api,
+    addonsStore
+  );
+  const backdropBoundary = useGetBackdropBoundary(
+    "syntax-highlighter-backdrop",
+    step === "customStory"
+  );
+
+  const project = useGetProject();
+  const isJavascript = project?.data?.language === "javascript";
+
+  const data = isJavascript
+    ? dataJavascript
+    : project?.data?.framework.name === "@storybook/nextjs"
+    ? dataTypescriptNextjs
+    : dataTypescript;
+
+  const copyWarningStory = () => {
+    const warningContent = data[4][0].content;
+    navigator.clipboard.writeText(
+      warningContent.replace("// Copy the code below", "")
+    );
+    setWarningStoryCopied(true);
+  };
 
-export function WriteStoriesModal({ onFinish }: { onFinish: () => void }) {
   return (
-    <Modal width={540} defaultOpen>
-      {({ Title, Description }) => (
-        <ModalContentWrapper data-chromatic="ignore">
-          <StorybookLogo />
-          <Title style={{ marginTop: 20 }}>
-            Create your first story (WORK IN PROGRESS)
-          </Title>
-          <Description>
-            Now it's your turn. <br />
-            See how easy it is to create your first story by following these
-            steps below.
-          </Description>
-          <Button style={{ marginTop: 4 }} onClick={onFinish}>
-            Go to story
-          </Button>
-        </ModalContentWrapper>
+    <Modal width={738} defaultOpen>
+      {({ Title, Description: DefaultDescription, Close }) => (
+        <ModalContent>
+          <div style={{ maxHeight: "450px" }}>
+            {data ? (
+              <SyntaxHighlighter
+                activeStep={
+                  step === "imports"
+                    ? 1
+                    : step === "meta"
+                    ? 2
+                    : step === "story"
+                    ? 3
+                    : step === "args"
+                    ? 4
+                    : step === "customStory"
+                    ? 5
+                    : 1
+                }
+                contents={data}
+                width="445px"
+              />
```

Comment:
Up to preference, but you could do something like this instead
```suggestion
              const stepIndex = {
                  imports: 1,
                  meta: 2,
                  story: 3,
                  args: 4,
                  customStory: 5,
                }

              // ...
              <SyntaxHighlighter
                activeStep={stepIndex[step] || 1}
                contents={data}
                width="445px"
              />
```

Reply:
I like this proposed change!